### ISSUE TITLE
Calendar Landing Page

### ISSUE THREAD

**Issue by sarah0arnold** [aemsites/clarkcountynv#1403]
**Block:** **Applies to this page: https://clarkcountynv.sharepoint.com/:w:/r/teams/AEM/_layouts/15/Doc.aspx?sourcedoc=%7BD865B41C-F2D3-449A-A9F2-2462DC039CF4%7D&file=calendar.docx&action=default&mobileredirect=true

**Design Links**:
- Calendar View: https://www.figma.com/design/T2bgro0OiuLqxvm1uLhJZ2/Clark-County-NV?node-id=1765-3436&t=TpjkoYVRPM18SCcg-0
- List View: https://www.figma.com/design/T2bgro0OiuLqxvm1uLhJZ2/Clark-County-NV?node-id=1765-3438&t=q74oP2ZERG1s0Kz3-0

**Live URL:** https://www.clarkcountynv.gov/calendar?view=month&day=01&month=07&year=2025

**User Story:** As a site visitor, I want the ability to view County events and meetings so that I can be informed and attend things in my community. 

**Requirements:** 
1. Remove/hide day & week view options
2. Remove/hide filters option
3. Updated search bar styling
- Add 'Search by keyword' field label with Body M text size and #232222 text color
- Placeholder text updated to 'Search' with Body M text size and #545454 text color
- Includes 'x' to clear search term and restore full calendar
- Replace button with primary filled button style with magnifying glass icon
- User can click 'enter' after typing search input or click the primary filled button to perform a search against the calendar
- Mobile search bar spans full screen width with the search button left aligned below the search bar
- **Search functionality to remain unchanged
4. User can toggle on desktop between calendar and list view and a parameter is added to the URL for their current view (only visible on desktop). "today" button sends user back to current month and year on click
5. The calendar/list view toggle is styled as buttons with the active view displaying with primary filled buttons and the inactive view with the primary outline button style
6. Calendar view Month & Year displays with 43px height text on desktop and 29px height on mobile and #963821 text color
7. User can click the outlined buttons with left/right arrows to advance forward/backward one month
- this displays inline with the month year on desktop and displays below the month year on mobile
8. Calendar view requirements: 
- Update all calendar events to the primary color with white text on calendar view
- start time displays bold and only displays minutes if minutes != 00
- If more than 3 events are added for a given day, then displays the (+ X events), which on click, sends the user to the list view automatically ***evaluating what is in place today, but ideal scenario is a fixed height for event display and then (+ X events) if more events on date than can display in fixed height
- Consistent height for all calendar rows
- Event bars will wrap to 2 lines as needed to show full event title (today's calendar only displays a single line per event)
- **Mobile ONLY displays list view**
9. List view requirements
- Displays a list of the month's events categorized by date
- Date has a solid primary background and displays with Heading M size text
- Events display in two columns with bold event start time in left column (time format only includes minutes if minutes != 00) and the event title hyperlinked in right column
10. On click of an event from either view, the event modal pops up: 
- remove hero time/date stamps
- 40x40px x with background color to close modal
- Update title within modal to primary color H3 style
- Update close button in footer of modal to 'Close Window' and update button styling for the close window and read more buttons. If only one button is present, then use primary filled button styling. If both buttons are present, then close window should display left with primary outline button styling and the read more button should display right with the primary filled button styling
11. Keyboard navigable: site visitor should be able to use their keyboard to interact with the calendar including: 
- Search for an event
- Change view
- Click 'Today'
- Navigate forward/backwards months in the calendar
- Navigate through the listed events including if a (+ X more) link is present
- Select an event to pop open the event modal
- Close the event modal by 'X' or close window in the footer of the event
- Navigate to any clickable items within the modal such as hyperlinks, map embeds, etc
- Click the 'read more' link in the footer of the event modal
- User should be trapped within the modal and not be able to navigate elsewhere on the site while the modal is open
12. Screen reader requirements
- Screen reader should announce date and time information in relation to an event
- Screen reader should announce event details within the modal window
- Screen reader should announce the 'x' to close modal window first

**Acceptance Criteria:** 
1. Given I am viewing the calendar on the county website, when these changes are implemented, then I should no longer see filters or day/week view options. 
2. Given I am viewing the calendar on the county website, when the calendar loads on desktop, then the search bar appears with a field title "Search by keyword" styled as Body M and #232222 text color.
3. Given I am viewing the calendar on the county website, when the calendar loads on desktop, then the search bar appears with placeholder text "Search" styled as Body M and #545454 text color.545454.
4. Given I am viewing the calendar on the county website, when the calendar loads, then the search bar includes an 'x' that will clear my search term and repopulate the calendar with all events.
5. Given I am viewing the calendar on the county website, when the calendar loads, then I am able to input a search term into the search field and either click enter or click the 'Search' button to perform my search. 
6. Given I am viewing the calendar on the county website, when on a mobile device, then the search bar appears full width and the 'Search' button displays left aligned below the search bar. 
7. Given I am viewing the calendar on desktop, when the calendar renders, then there is a list view/calendar view toggle and the default view on calendar page load is calendar view.
8. Given I am viewing the calendar on mobile, when the calendar renders, then the default view is list and I do not see a toggle to change view. 
9. Given I am viewing the calendar on desktop, when the calendar renders, then the calendar/list view toggle is styled as buttons with the active view having the primary filled button styling and the inactive view having the primary outlined button styling.
10. Given I am viewing the calendar, when I click 'Today', then I am taken back to the current month/year calendar. 
11. Given I am viewing calendar, when the page renders, then the month/year title displays with 43px height on desktop and 29px height on mobile.
12. Given I am viewing the calendar, when I click a left/right arrow button, then I am advanced forward/backward by one month in the calendar. 
13. Given I am viewing the calendar, when viewing on desktop, then the arrows and 'today' button display inline with the calendar month year title. 
14. Given I am viewing the calendar, when viewing on mobile, then the arrows and 'today' button display just below the calendar month year title. 
15. Given I am viewing the calendar on desktop, when the view is set to calendar, then the events display with primary colored bars with white text across their respective dates.
16. Given I am viewing the calendar on desktop in calendar view, when there are more than 3 events, then a hyperlink displays with '+ X events' below the first two events for that date. 
17. Given there are more than 3 events for a date, when I click the '+ X events' link, then I am taken to list view for that date so I can see a full list of all events. 
18. Given I am viewing the calendar, when an event start time is the top of an hour, then it only displays with the hour and AM/PM for the event time. 
19. Given I am viewing the calendar, when an event start time is not the top of an hour, then it displays with hour+minute information and AM/PM for the event. 
20. Given I am viewing the calendar, when on list view, then the selected months events display, categorized by date. 
21. Given I am viewing list view, when the view renders, then the date has a primary background color bar with white text with Heading M styling. 
22. Given I am viewing list view, when the view renders, then events for a specific date are styled with general hyperlink styling. 
23. Given I am viewing calendar, when I select a specific event, then the modal pops up for the event. 
24. Given I am viewing an event popup modal, when it renders, then the date/timestamp are no longer in the hero banner. 
25. Given I am viewing an event popup modal, when it renders, then the 'X' close icon displays in the top right corner sized 40x40px with background color for visibility and touch target size. 
26. Given I am viewing an event popup modal, when it renders, then the heading is sized H3 and primary colored. 
27. Given I am viewing an event popup modal, when it renders, then the button in the footer to close the modal is updated to text 'Close Window'.
28. Given I am viewing an event popup modal, when there is not a 'read more' link included for an event, then the close window button displays with primary filled button styling.
29. Given I am viewing an event popup modal, when there is a 'read more' link included for an event then two buttons display in the footer with the close window button to the left with primary outline styling and the read more button displays right with the primary filled button styling.
30. Given I am viewing the calendar, when I am keyboard navigating, then I am able to navigate to all clickable items in calendar, perform a search, and open a desired event. 
31. Given I am viewing an event popup modal, when keyboard navigating, then I am able to navigate to any hyperlink or map embed with the event modal as well as close the modal when finished. 
32. Given I am viewing an event popup modal, when keyboard navigating, then I should be trapped within the modal until I close the modal. 
33. Given I am viewing the calendar, when using a screen reader, then the events are read in logical order and I can easily understand the event details. 
34. Given I am viewing an event popup modal, when using a screen reader, then the 'x' close icon is announced first and event details are read in logical order.

**Comment by aaronstewart-dev** [comment]
@beccasuchanek Resolved calendar issues, latest on develop branch. Assigning QA label.

**Comment by sarah0arnold** [comment]
Calendar landing page reviewed in UAT session with project team 8/4/25 and passes with no feedback. Closing ticket
